common/flatpak-oci-registry.c

```diff
@@ -456,13 +505,152 @@ verify_oci_version (GBytes *oci_layout_bytes, gboolean *not_json, GCancellable *
   return TRUE;
 }
 
+/*
+ * This is a small helper to handle cleanup for thetemporary directory
+ * we unpack an archive into. We can't use GLnxTmpDir for this because
+ * the dfd usage isn't compatible with the way we use archive_write_disk
+ */
+typedef struct
+{
+  gboolean unlink;
+  char *path;
+} TemporaryDirectory;
+
+static void
+temporary_directory_free (TemporaryDirectory *self)
+{
+  if (self->path && self->unlink)
+    glnx_shutil_rm_rf_at (AT_FDCWD, self->path, NULL, NULL);
+
+  g_free (self->path);
+  g_free (self);
+}
+
+G_DEFINE_AUTOPTR_CLEANUP_FUNC (TemporaryDirectory, temporary_directory_free)
+
+static TemporaryDirectory *
+temporary_directory_new (GError **error)
+{
+  g_autoptr(TemporaryDirectory) self = g_new0 (TemporaryDirectory, 1);
+  self->path = g_dir_make_tmp ("oci-XXXXXX", error);
```

Comment:
Isn't this /tmp, and should we not use /var/tmp here, as the archive may be large?

Reply:
Hmm, yeah - or actually should be FLATPAK_DOWNLOAD_TMPDIR as introduced in #5992 - I've updated it to be `/var/tmp` for now.

Reply:
Getting https://github.com/flatpak/flatpak/pull/5992 merged would be nice.

Reply:
Landed #5992.